DOL.WHD.Section14c.Api/Controllers/AttachmentController.cs

```diff
@@ -49,15 +50,22 @@ public AttachmentController(IAttachmentService attachmentService, IIdentityServi
             }
 
             var filesReadToProvider = await Request.Content.ReadAsMultipartAsync(new RestrictedMultipartMemoryStreamProvider());
+
             var files = new List<Domain.Models.Submission.Attachment>();
+            var allowedMaximumContentLength = AppSettings.Get<int>("AllowedMaximumContentLength");
             foreach (var stream in filesReadToProvider.Contents)
-            {
-                if (stream.Headers.ContentLength == 0)
-                    return BadRequest("Invalid file.");
+            {               
+               // The code that handle the max allowed length at the IIS level within web.config.
+               // as well as within the httpRuntime attribute found in the system.web section
+               var bytes = await stream.ReadAsByteArrayAsync();
 
-                using (var memoryStream = new MemoryStream())
+               if( bytes.Length <1 || bytes.Length  > allowedMaximumContentLength)
                 {
-                    await stream.CopyToAsync(memoryStream);
+                    return BadRequest("Invalid file size.");
+                }
```

Comment:
My only concern with this approach is that whole files will have to be read into memory, which could put memory pressure/exhaustion on the system.  But since there's an IIS config to prevent bodies over a certain size, then this should not grow unbounded.

@mmurthydol This risk belongs to WHD.  I'm fine with it if you are!

Reply:
@mgwalker @dciceman Thanks for raising the issue Greg. Looks like David fixed above code snippet with MultiPart related code. I just reviewed it - seems fine. Will however create a performance defect in fileRepository.cs to work with byte[] overload rather than a MemoryStream - similar issue as raised above. #353
https://github.com/18F/dol-whd-14c/issues/353